kotlin/spatial-operations/src/main/java/com/esri/arcgisruntime/sample/spatialoperations/MainActivity.kt

```diff
@@ -0,0 +1,217 @@
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.spatialoperations
+
+import android.graphics.Color
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geometry.Geometry
+import com.esri.arcgisruntime.geometry.GeometryEngine
+import com.esri.arcgisruntime.geometry.Part
+import com.esri.arcgisruntime.geometry.PartCollection
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.PointCollection
+import com.esri.arcgisruntime.geometry.Polygon
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val inputGeometryOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+  private val resultGeometryOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  // simple black (0xFF000000) line symbol for outlines
+  private val lineSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 1f)
+  private val resultFillSymbol = SimpleFillSymbol(
+    SimpleFillSymbol.Style.SOLID, -0x16e0e1, lineSymbol
+  )
+  private var inputPolygon1: Polygon? = null
+  private var inputPolygon2: Polygon? = null
+
+  // The spatial operation switching menu items.
+  private var noOperationMenuItem: MenuItem? = null
+  private var intersectionMenuItem: MenuItem? = null
+  private var unionMenuItem: MenuItem? = null
+  private var differenceMenuItem: MenuItem? = null
+  private var symmetricDifferenceMenuItem: MenuItem? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    mapView.apply {
+      // create ArcGISMap with topographic basemap
+      map = ArcGISMap(Basemap.createLightGrayCanvas())
+
+      // create graphics overlays to show the inputs and results of the spatial operation
+      graphicsOverlays.add(inputGeometryOverlay)
+      graphicsOverlays.add(resultGeometryOverlay)
+
+      // create input polygons and add graphics to display these polygons in an overlay
+      createPolygons()
+
+      // center the map view on the input geometries
+      val envelope = GeometryEngine.union(inputPolygon1, inputPolygon2).extent
+      setViewpointGeometryAsync(envelope, 20.0)
+    }
+
+  }
+
+  private fun showGeometry(resultGeometry: Geometry) {
+    // add a graphic from the result geometry, showing result in red (0xFFE91F1F)
+    Graphic(resultGeometry, resultFillSymbol).let {
+      resultGeometryOverlay.graphics.add(it)
```

Comment:
This pattern might be a bit hard to read...usually the stuff in the scoping function is stuff happening to the receiver rather than the receiver being added to something else. Maybe move towards something like:

```
    val graphic = Graphic(resultGeometry, resultFillSymbol).also {
      it.isSelected = true
    }
    resultGeometryOverlay.graphics.add(graphic)
```